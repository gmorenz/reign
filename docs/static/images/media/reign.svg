<?xml version="1.0" encoding="UTF-8"?>
<svg version="1.1" viewBox="0 0 224 224" xmlns="http://www.w3.org/2000/svg" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <metadata>
    <rdf:RDF>
      <cc:Work rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"/>
        <dc:title/>
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g stroke-width=".32">
    <path fill="#222222" d="m0 112v-112h224v224h-224z"/>
    <path fill="#ab923a" d="m157.77 134.4-20.539 22.207-6.2065-19.533-16.771 20.027-16.675-20.141-6.4537 20.233-20.743-21.833c-4.4543 7.9357-7.2165 14.554-11.826 23.019l-27.075-71.884 37.666 39.359 10.202-19.404 16.023 19.145 18.761-60.954 18.953 61.34c16.429-19.783 0 0 16.399-19.754l9.5581 18.874 33.834-38.22-22.596 71.839zm-28.5-3.7736-15.057-48.706-15.058 48.895 15.137 18.288zm-35.751 1.2154-13.437-16.427-7.1207 14.646 15.805 17.018zm61.896-2.5289-7.001-13.878-13.372 16.715 5.1662 14.482zm-88.792 1.8254-21.028-22.542 13.666 37.06zm114.11-22.832-19.375 21.754 7.9404 14.856zm-105.09-5.7257c-2.9221-1.2718-2.6787-5.2577 0.38277-6.2681 3.7289-1.2306 5.997 4.0301 2.6242 6.0865-1.2222 0.74517-1.6522 0.77113-3.0069 0.18153zm74.219-0.16426c-1.2604-0.76851-1.7004-1.5567-1.7004-3.0457 0-2.2131 2.7756-3.9371 4.72-2.9316 1.2435 0.64303 1.6783 1.2192 1.8814 2.4928 0.23604 1.4804-0.31615 2.7129-1.5534 3.4672-1.3134 0.80074-2.0563 0.80458-3.3476 0.0173zm-125.06-19.031c-2.8412-1.2735-3.8691-4.345-2.4107-7.2037 1.2137-2.3791 4.8884-3.2732 7.2075-1.7537 2.0585 1.3488 2.7774 4.9102 1.4354 7.1113-1.155 1.8944-4.147 2.7807-6.2322 1.8461zm171.68 0.11322c-2.8789-1.2106-4.1308-4.9159-2.4802-7.3411 1.2288-1.8055 2.2894-2.3861 4.3585-2.3861 2.1124 0 3.2511 0.58526 4.2158 2.1667 2.0464 3.3549 0.0242 7.5126-3.7878 7.788-0.82858 0.05986-1.8664-0.04251-2.3064-0.22749zm-92.739-36.234 10.531-20.704 10.264 20.392-10.245 10.648z"/>
  </g>
</svg>
