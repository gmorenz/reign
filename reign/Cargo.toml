[package]
name = "reign"
version = "0.1.2"
authors = [
	"Pavan Kumar Sunkara <pavan.sss1991@gmail.com>"
]
edition = "2018"
description = "Opinionated Fullstack Web Framework in Rust"
repository = "https://github.com/pksunkara/reign"
keywords = ["fullstack", "web", "framework", "reign"]
categories = []
license = "MIT OR Apache-2.0"
readme = "README.md"
autoexamples = false

[[bin]]
name = "reign"
path = "src/cli/main.rs"
required-features = ["cli"] # remove the need for this

[lib]
name = "reign"
path = "src/lib.rs"

[dependencies]
# common deps
log = "0.4.8"

# lib deps
reign_boot = { path = "./reign_boot", version = "0.1.2", optional = true }
reign_derive = { path = "./reign_derive", version = "0.1.2", optional = true }
reign_router = { path = "./reign_router", version = "0.1.2", optional = true }
reign_view = { path = "./reign_view", version = "0.1.2", optional = true }

# cli deps
clap = { version = "3.0.0-beta.1", optional = true }

[dev-dependencies]
version-sync = "0.9.0"

[workspace]
members = [
	"reign_boot",
	"reign_derive",
	"reign_router",
	"reign_view",

	"development/test_examples",
]
default-members = [
	".",
	"reign_boot",
	"reign_derive",
	"reign_router",
	"reign_view",
]

[features]
default = ["view-gotham", "router-gotham", "framework"]
cli = ["clap"]
doc = []
framework = ["reign_boot"]
view = ["reign_view", "reign_derive/view"]
view-actix = ["reign_view/view-actix", "reign_derive/view-actix", "view"]
view-gotham = ["reign_view/view-gotham", "reign_derive/view-gotham", "view"]
view-tide = ["reign_view/view-tide", "reign_derive/view-tide", "view"]
view-warp = ["reign_view/view-warp", "reign_derive/view-warp", "view"]
json = ["reign_view/json", "reign_derive/json", "view"]
router = ["reign_router", "reign_derive/router"]
router-actix = ["reign_router/router-actix", "reign_derive/router-actix", "router"]
router-gotham = ["reign_router/router-gotham", "reign_derive/router-gotham", "router"]
router-tide  = ["reign_router/router-tide", "reign_derive/router-tide", "router"]
router-warp = ["reign_router/router-warp", "reign_derive/router-warp", "router"]

[package.metadata.docs.rs]
all-features = true
