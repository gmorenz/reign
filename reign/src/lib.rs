#![cfg_attr(feature = "doc", feature(external_doc))]
#![doc(html_logo_url = "https://reign.rs/images/media/reign.svg")]
#![doc(html_root_url = "https://docs.rs/reign/0.1.2")]
#![cfg_attr(feature = "doc", doc(include = "../README.md"))]

pub use log;

pub mod prelude;

#[cfg(feature = "framework")]
pub use reign_boot::boot;
#[cfg(feature = "router")]
pub use reign_router as router;
#[cfg(feature = "view")]
pub use reign_view as view;
