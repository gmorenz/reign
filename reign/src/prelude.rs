#[cfg(any(feature = "view", feature = "router"))]
pub use reign_derive::*;
