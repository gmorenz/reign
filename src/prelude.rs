#[cfg(any(
    feature = "framework",
    feature = "view",
    feature = "router-backend",
    feature = "model-postgres",
    feature = "hot-reload"
))]
pub use reign_derive::*;
#[cfg(feature = "router-backend")]
pub use reign_router::{Error, Request, Response};
#[cfg(feature = "view-backend")]
pub use reign_view::redirect;
