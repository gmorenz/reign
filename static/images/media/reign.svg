<?xml version="1.0" encoding="UTF-8"?>
<svg version="1.1" viewBox="0 0 224 224" xmlns="http://www.w3.org/2000/svg" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <metadata>
    <rdf:RDF>
      <cc:Work rdf:about="">
        <dc:format>image/svg+xml</dc:format>
        <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"/>
        <dc:title/>
      </cc:Work>
    </rdf:RDF>
  </metadata>
  <g stroke-width=".32">
    <path fill="#222222" d="m0 112v-112h224v224h-224z"/>
    <path fill="#ab923a" d="m164.03 146.56c-3.3271-6.6875-6.1451-12.159-6.2623-12.159-0.11715 0-0.55459 0.396-0.97208 0.88-4.4757 5.1887-7.3873 8.5362-10.229 11.76-1.8614 2.112-4.6738 5.337-6.2498 7.1666-2.8332 3.2894-2.8678 3.3163-3.0888 2.4-0.12295-0.50965-1.0354-3.5186-2.0278-6.6866-1.629-5.2006-3.0543-9.8729-3.6542-11.979-0.12207-0.42853-0.35809-0.81897-0.5245-0.86764-0.1664-0.0487-1.5266 1.4298-3.0226 3.2854s-3.872 4.787-5.28 6.5142c-1.408 1.7271-3.8508 4.7393-5.4284 6.6938-1.5776 1.9544-2.9456 3.5447-3.04 3.534-0.0944-0.0107-2.0436-2.3332-4.3316-5.1611-2.288-2.8279-5.6-6.9008-7.36-9.0508-1.76-2.1501-3.5455-4.3636-3.9679-4.9189-0.42232-0.55532-0.87916-1.0097-1.0152-1.0097-0.13603 0-0.39282 0.468-0.57064 1.04-0.31951 1.0277-1.8725 6.1605-3.4613 11.44-0.45021 1.496-1.1793 3.8523-1.6202 5.2363l-0.80158 2.5163-1.0383-1.0763c-0.57104-0.59196-2.3898-2.5194-4.0416-4.2833-1.6519-1.7638-5.1005-5.4358-7.6635-8.16-2.5631-2.7242-5.3548-5.709-6.2038-6.633-0.849-0.924-1.6572-1.68-1.796-1.68s-2.8191 5.1744-5.9562 11.499c-3.1371 6.3243-5.7786 11.508-5.8699 11.52-0.09133 0.0117-0.5774-1.0947-1.0802-2.4587-0.50277-1.364-1.9238-5.144-3.1579-8.4-1.2341-3.256-2.8859-7.648-3.6707-9.76-0.78482-2.112-1.5402-4.128-1.6786-4.48-0.13843-0.352-0.86135-2.296-1.6065-4.32s-1.4704-3.968-1.6116-4.32c-0.37004-0.92207-4.031-10.662-5.3523-14.24-0.61748-1.672-1.5897-4.264-2.1604-5.76s-1.4323-3.8-1.9145-5.12c-0.48224-1.32-0.99132-2.688-1.1313-3.04-1.6064-4.04-3.7877-9.9088-3.7112-9.9853 0.10867-0.10867 1.3745 1.2206 13.321 13.99 2.376 2.5395 5.407 5.7734 6.7356 7.1864 1.3286 1.4131 3.5606 3.7974 4.96 5.2984 8.1927 8.7877 12.434 13.108 12.649 12.884 0.13402-0.13952 2.4306-4.6233 5.1034-9.964 2.6729-5.3406 4.9673-9.5886 5.0988-9.44 0.1315 0.14865 1.5159 1.8543 3.0764 3.7903 1.5605 1.936 3.9013 4.816 5.2018 6.4 1.3004 1.584 2.6009 3.168 2.8899 3.52 3.7762 4.5996 4.6402 5.5668 4.8547 5.4343 0.13837-0.0855 0.4991-0.98322 0.80163-1.9949 0.30252-1.0117 1.1214-3.7114 1.8196-5.9994s1.6835-5.528 2.1894-7.2c1.0227-3.3799 3.096-10.153 4.4567-14.56 0.48913-1.584 1.3502-4.464 1.9136-6.4 0.56336-1.936 1.4304-4.816 1.9266-6.4 0.49631-1.584 1.4269-4.608 2.068-6.72 2.2454-7.3975 3.4138-11.204 3.585-11.68 0.095-0.264 0.95819 2.112 1.9183 5.28 0.96011 3.168 2.254 7.416 2.8754 9.44s1.6312 5.336 2.2441 7.36 1.6037 5.264 2.2018 7.2c0.59812 1.936 1.5898 5.176 2.2037 7.2 2.2258 7.3379 3.3712 11.092 4.1981 13.76 0.46367 1.496 0.89256 2.936 0.95307 3.2 0.0605 0.264 0.46759 1.5902 0.90461 2.947 0.43702 1.3569 0.89039 2.9032 1.0075 3.4363 0.39412 1.7944 0.49645 1.734 3.764-2.2234 0.94457-1.144 2.9303-3.592 4.4126-5.44 1.4824-1.848 4.0393-4.9831 5.6821-6.9669l2.9869-3.6069 4.6991 9.4467c2.5845 5.1957 4.771 9.438 4.859 9.4273 0.088-0.0106 1.4196-1.4865 2.9592-3.2798 2.7629-3.2182 4.0843-4.7278 6.094-6.962 1.372-1.5252 4.0992-4.6512 6.9842-8.0053 9.6831-11.257 18.179-20.792 17.797-19.973-0.082 0.176-0.97191 2.984-1.9775 6.24-1.0056 3.256-2.0895 6.712-2.4088 7.68s-1.0223 3.2-1.5622 4.96c-0.53993 1.76-1.8678 6.008-2.9509 9.44s-2.1526 6.888-2.3768 7.68c-0.22417 0.792-0.79677 2.664-1.2724 4.16-0.47567 1.496-1.805 5.672-2.954 9.28s-2.2737 7.208-2.4992 8c-0.72338 2.5404-2.7475 9.0318-3.1851 10.215-0.23304 0.62997-0.60076 1.8294-0.81714 2.6654-0.21639 0.836-0.48272 1.5196-0.59185 1.519-0.10912-6.4e-4 -2.9206-5.4725-6.2477-12.16zm-47.84 0.40097c0.92907-1.188 3.0612-3.816 4.7381-5.84s3.1631-3.824 3.3027-4c0.13962-0.176 1.3276-1.616 2.6399-3.2 1.3123-1.584 2.3911-3.0661 2.3973-3.2936 6e-3 -0.22749-0.54517-2.1715-1.2252-4.32-1.5067-4.7603-3.1705-10.14-4.5233-14.626-1.8733-6.2123-3.5338-11.656-4.6364-15.2-0.60233-1.936-1.4707-4.816-1.9297-6.4-1.7606-6.0756-2.4611-8.16-2.7426-8.16-0.16037 0-0.29158 0.17824-0.29158 0.39609 0 0.21785-0.50817 2.0165-1.1293 3.997-0.6211 1.9805-1.6243 5.2223-2.2293 7.2039s-1.5511 5.043-2.1025 6.803-1.184 3.848-1.4058 4.64c-0.80389 2.8698-4.6703 15.572-6.9301 22.768l-0.96969 3.0876 3.3033 3.9976c3.9454 4.7747 9.3283 11.366 10.663 13.058 0.528 0.66897 1.0548 1.2238 1.1708 1.2328 0.11593 9e-3 0.97094-0.95548 1.9-2.1435zm-25.424-5.5352c0.9439-3.0717 1.9488-6.4844 2.2332-7.5839l0.51702-1.9991-0.80239-1.0409c-0.44132-0.5725-1.1226-1.4009-1.5139-1.8409-0.92159-1.0362-6.5819-8.0093-7.8903-9.7202-0.5551-0.72589-1.5089-1.8836-2.1197-2.5726l-1.1104-1.2528-3.5603 7.1223c-1.9582 3.9173-3.5603 7.303-3.5603 7.5239 0 0.51704 1.1799 1.8263 11.36 12.606 1.232 1.3045 2.672 2.8465 3.2 3.4267 0.528 0.58018 1.0885 1.0238 1.2455 0.9859 0.157-0.0379 1.0578-2.5822 2.0017-5.6538zm57.306-3.7284 7.34-8.3835-3.0844-6.1765c-1.6964-3.397-3.2717-6.5196-3.5005-6.939l-0.41603-0.76256-0.76465 0.92256c-0.42055 0.50741-1.0892 1.3334-1.486 1.8356-0.39672 0.50216-1.6573 2.0239-2.8013 3.3817-1.144 1.3578-2.44 2.9516-2.88 3.5419-0.44 0.59029-1.844 2.3298-3.12 3.8655-1.276 1.5357-2.32 2.9612-2.32 3.1676 0 0.20647 0.48747 1.9132 1.0833 3.7928s1.6397 5.1775 2.3198 7.3287c1.1558 3.6558 1.271 3.8753 1.7632 3.36 0.28964-0.30321 3.8296-4.3239 7.8666-8.9348zm-85.025 0.59269 3.5726-7.1508-0.91031-0.98691c-0.50067-0.5428-1.4143-1.5159-2.0303-2.1625s-1.9655-2.0966-2.9989-3.2222c-1.6982-1.8498-7.3098-7.8265-12.619-13.44-1.1099-1.1735-2.2213-2.402-2.4698-2.7302-0.93516-1.2347-0.4836 0.2134 2.7077 8.6834 0.92838 2.464 2.2571 5.992 2.9526 7.84 0.69555 1.848 1.6754 4.44 2.1774 5.76 0.50202 1.32 1.4229 3.768 2.0464 5.44s1.6804 4.5054 2.3487 6.2963c0.66828 1.791 1.313 3.159 1.4328 3.04 0.11974-0.11899 1.8254-3.4342 3.7903-7.3672zm107.31 3.7718c0.4345-1.3095 0.9311-2.8849 1.1036-3.5009 0.33065-1.181 2.7092-8.8486 4.9138-15.84 0.72146-2.288 1.8081-5.744 2.4147-7.68s1.3227-4.2133 1.5912-5.0607c0.26858-0.8474 0.42863-1.6004 0.35567-1.6734-0.073-0.073-2.8596 2.9883-6.1926 6.8028s-6.492 7.4015-7.02 7.971c-0.528 0.56958-1.4892 1.6576-2.1361 2.4179-0.64684 0.76028-1.8173 2.0977-2.601 2.9721l-1.4249 1.5898 3.8376 7.6651c2.7164 5.4257 3.915 7.5266 4.1028 7.1911 0.14588-0.26068 0.62074-1.5454 1.0552-2.8549zm-94.713-39.481c-2.9221-1.2718-2.6787-5.2577 0.38277-6.2681 3.7289-1.2306 5.997 4.0301 2.6242 6.0865-1.2222 0.74517-1.6522 0.77113-3.0069 0.18153zm74.219-0.16426c-1.2604-0.76851-1.7004-1.5567-1.7004-3.0457 0-2.2131 2.7756-3.9371 4.72-2.9316 1.2435 0.64303 1.6783 1.2192 1.8814 2.4928 0.23604 1.4804-0.31615 2.7129-1.5534 3.4672-1.3134 0.80074-2.0563 0.80458-3.3476 0.0173zm-125.06-19.031c-2.8412-1.2735-3.8691-4.345-2.4107-7.2037 1.2137-2.3791 4.8884-3.2732 7.2075-1.7537 2.0585 1.3488 2.7774 4.9102 1.4354 7.1113-1.155 1.8944-4.147 2.7807-6.2322 1.8461zm171.68 0.11322c-2.8789-1.2106-4.1308-4.9159-2.4802-7.3411 1.2288-1.8055 2.2894-2.3861 4.3585-2.3861 2.1124 0 3.2511 0.58526 4.2158 2.1667 2.0464 3.3549 0.0242 7.5126-3.7878 7.788-0.82858 0.05986-1.8664-0.04251-2.3064-0.22749zm-87.571-31.066-5.1681-5.1681 5.166-10.352c2.8413-5.6935 5.2557-10.352 5.3654-10.352 0.1921 0 10.055 19.593 10.264 20.392 0.09748 0.37097-9.7902 10.648-10.245 10.648-0.11821 0-2.5406-2.3257-5.3831-5.1681z"/>
  </g>
</svg>
