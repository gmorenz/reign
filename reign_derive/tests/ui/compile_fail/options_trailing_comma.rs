use reign::prelude::*;

fn test() {
    render!(pages::home, status = 100,);
}

fn main() {}
