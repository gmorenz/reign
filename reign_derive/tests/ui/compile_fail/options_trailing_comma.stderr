error: unexpected end of input, expected identifier
 --> $DIR/options_trailing_comma.rs:4:5
  |
4 |     render!(pages::home, status = 100,);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `render` (in Nightly builds, run with -Z macro-backtrace for more info)
